### PR TITLE
fix: Save user details in order confirmation step

### LINKED ISSUE #2945: Orders: Buyer info is missing after successful order
image

### PR DESCRIPTION
**Pull request by uds5501**
Fixes #2945

#### Changes proposed in this pull request:

- add a order.user.save() method to save the user model

### CODE REVIEW

File: app/controllers/orders/new.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ export default Controller.extend({
       try {
         this.set('isLoading', true);
         let order = data;
+        await order.user.get('save');
```
Comment (CosmicCoder96) [inline app/controllers/orders/new.js]: `this.get('authManager.currentUser').save()`
and this save should be conditional --> only if the value for the name  was previously null, and now there is a name provided or edited, we make this extra call.

Comment (uds5501) [reply app/controllers/orders/new.js]: @CosmicCoder96 Okay I will test this out

Comment (uds5501) [reply app/controllers/orders/new.js]: @CosmicCoder96 Okay perfect, it works. Pushing the changes soon

**Comment by uds5501** [comment]
@shreyanshdwivedi @CosmicCoder96 Please review now

File: app/controllers/orders/new.js [hunk 2]
```diff
@@ -6,6 +6,12 @@ export default Controller.extend({
       try {
         this.set('isLoading', true);
         let order = data;
+        let current_user = this.get('authManager.currentUser');
```
Comment (CosmicCoder96) [inline app/controllers/orders/new.js]: current_user and order.user are the same things, just use a check for dirtyAttributes for the user model.

Comment (uds5501) [reply app/controllers/orders/new.js]: @CosmicCoder96 on it

Comment (uds5501) [reply app/controllers/orders/new.js]: Completed the required changes

**Pull request merged**